docs/developer.rst

```diff
@@ -0,0 +1,204 @@
+=================
+Developer's guide
+=================
+
+Required Modules
+================
+
+If you are installing from source you will need a Python 2.6+ installation and the following modules:
+
+::
+
+  setuptools
+  xlrd
+  wxPython 
+  Sphinx
+  
+  
+In case you find errors with wxPython, try `installing the wheel`_ from a third party distributor
+
+Setting up servers
+==================
+
+You need to install all the database infrastructures to enable local testing.
+
+::  
+  
+  SQlite
+  MySQL 5.6
+  PostgreSQL 9.1
+  MSAccess
+
+You will also need the following modules:
+
+::
+  
+  mysqldb (MySQL)
+  psycopg2 (PostgreSQL)
+  pyodbc (MS Access).
+  py2app (Mac)
+  py2exe (Windows)
+  
+Style Guide for Python Code
+===========================
+
+Run ``pep8`` on the given file to make sure the file follows the right style.
+In some cases we do tend to work outside the ``pep8`` requirements.
+The compromise on ``pep8``  may be a result of enforcing better code readability.
+In some cases ``pep`` shows errors for long lines, but that can be ignored.
+
+``pep8 pythonfile.py``
+
+Testing
+=======
+
+Follow these instructions to run a complete set of test for any branch
+Clone the branch you want to test
+
+two ways of installing the program using the `setup tools`_.
+
+we can either install from source as 
+
+.. code-block:: bash
+
+  $ pip setup.py install
+
+or install in development mode.
+
+.. code-block:: bash
+
+  $  pip setup.py develop
+
+For more about `installing`_ refer to the python setuptool `documentation`_.
+
+you can also install from Git.
+
+.. code-block:: bash
+
+  # Local repository
+  pip install git+file:///path/to/your/git/repo #  test a PIP package located in a local git repository
+  pip install git+file:///path/to/your/git/repo@branch  # checkout a specific branch by adding @branch_name at the end
+  
+  # Remote github repository
+  pip install git+git://github.com/myuser/myproject  #  package from a github repository
+  pip install git+git://github.com/myuser/myproject@my_branch # github repository Specific branch
+
+Running tests locally
+^^^^^^^^^^^^^^^^^^^^^^^
+
+To run tests we use pytest.
+From the source top level directory, run
+
+.. code-block:: sh
+
+  $   py.test
+
+
+In case we want to run tests on a specific test category, we add the path of the test module, py.test [path]
+
+.. code-block:: sh
+
+  $   py.test .\test\test_retriever.py
```

Comment:
This should probably be `./test/test_retriever.py`. As written it won't run on Linux. I think, but can't remember for sure that Windows will often handle `/` in place of `\`.